contracts/contracts/strategies/ConvexEthMetaStrategy.sol

```diff
@@ -21,75 +21,141 @@ contract ConvexEthMetaStrategy is InitializableAbstractStrategy {
+            int256(balancesAfter[oethCoinIndex]);
+
+        if (diffBefore <= 0) {
+            // If the pool was originally imbalanced in favor of OETH, then
+            // we want to check that the pool is now more balanced
+            require(diffAfter <= 0, "OTokens overshot peg");
+            require(diffBefore < diffAfter, "OTokens balance worse");
+        }
+        if (diffBefore >= 0) {
+            // If the pool was originally imbalanced in favor of ETH, then
+            // we want to check that the pool is now more balanced
+            require(diffAfter >= 0, "Assets overshot peg");
+            require(diffAfter < diffBefore, "Assets balance worse");
+        }
+    }
+
+    // Used to circumvent the stack too deep issue
+    struct ConvexEthMetaConfig {
         address cvxDepositorAddress; //Address of the Convex depositor(AKA booster) for this pool
-        address oethAddress; //Address of OETH token
         address cvxRewardStakerAddress; //Address of the CVX rewards staker
-        address curvePoolLpToken; //Address of metapool LP token
         uint256 cvxDepositorPTokenId; //Pid of the pool referred to by Depositor and staker
+        address oethAddress; //Address of OETH token
+        address wethAddress; //Address of WETH
     }
 
-    constructor(BaseStrategyConfig memory _stratConfig)
-        InitializableAbstractStrategy(_stratConfig)
-    {}
+    constructor(
+        BaseStrategyConfig memory _baseConfig,
+        ConvexEthMetaConfig memory _convexConfig
+    ) InitializableAbstractStrategy(_baseConfig) {
+        lpToken = IERC20(_baseConfig.platformAddress);
+        curvePool = ICurveETHPoolV1(_baseConfig.platformAddress);
+
+        cvxDepositorAddress = _convexConfig.cvxDepositorAddress;
+        cvxRewardStaker = IRewardStaking(_convexConfig.cvxRewardStakerAddress);
+        cvxDepositorPTokenId = _convexConfig.cvxDepositorPTokenId;
+        oeth = IERC20(_convexConfig.oethAddress);
+        weth = IWETH9(_convexConfig.wethAddress);
+    }
 
     /**
      * Initializer for setting up strategy internal state. This overrides the
      * InitializableAbstractStrategy initializer as Curve strategies don't fit
      * well within that abstraction.
      * @param _rewardTokenAddresses Address of CRV & CVX
-     * @param _assets Addresses of supported assets. MUST be passed in the same
-     *                order as returned by coins on the pool contract, i.e.
-     *                WETH
-     * @param initConfig Various addresses and info for initialization state
+     * @param _assets Addresses of supported assets. eg WETH
      */
     function initialize(
         address[] calldata _rewardTokenAddresses, // CRV + CVX
-        address[] calldata _assets,
-        address[] calldata _pTokens,
-        InitializeConfig calldata initConfig
+        address[] calldata _assets // WETH
     ) external onlyGovernor initializer {
         require(_assets.length == 1, "Must have exactly one asset");
```

Comment:
Should we require that the one asset be eth? Since that's hardcoded elsewhere in this file.

Reply:
Yes, the asset must be WETH. I've added this check against the immutable `weth` variable. This works for the unit test using mocks.